tests/unit_tests/components/test_tag.py

```diff
@@ -0,0 +1,52 @@
+import pytest
+
+from app.components.tag import Tag, TagColor
+
+
+class TestTag:
+    def test_init_valid_text_and_color(self):
+        tag = Tag("Active", TagColor.GREEN)
+
+        assert tag.text == "Active"
+        assert tag.color == TagColor.GREEN
+
+    def test_init_empty_text_raises_error(self):
+        with pytest.raises(ValueError, match="Text must be a non-empty string"):
+            Tag("", TagColor.RED)
+
+    def test_init_whitespace_text_raises_error(self):
+        with pytest.raises(ValueError, match="Text must be a non-empty string"):
+            Tag("   ", TagColor.BLUE)
+
+    def test_init_none_text_raises_error(self):
+        with pytest.raises(ValueError, match="Text must be a non-empty string"):
+            Tag(None, TagColor.YELLOW)
+
+    def test_init_non_string_text_raises_error(self):
+        with pytest.raises(ValueError, match="Text must be a non-empty string"):
+            Tag(123, TagColor.PURPLE)
+
+    def test_init_invalid_color_raises_error(self):
+        with pytest.raises(ValueError, match="Color must be a TagColor enum value"):
+            Tag("Valid text", "invalid-color")
+
```

Comment:
can we add a test that shows multiple tags at the same time e.g inactive & hold payments/